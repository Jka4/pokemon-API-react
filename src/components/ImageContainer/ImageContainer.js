import React from 'react';
import { useImage } from 'react-image'
import ErrorBoundary from '@ErrorBoundary';

const ImageContainer = (props) => {
  const ImageWrapper = (props) => {
    const { url, cn } = props.props
    const { src } = useImage({
      srcList: url,
    })

    return (
      <React.Fragment>
        <img loading="lazy" src={src} alt={cn !== undefined ? cn : ''} className={cn !== undefined ? cn : ''} />
      </React.Fragment>
    )
  }

  return (
    <ErrorBoundary >
      <ImageWrapper props={props} />
    </ErrorBoundary>
  )
}

export default ImageContainer;
