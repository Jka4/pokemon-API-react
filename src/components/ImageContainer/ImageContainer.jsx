import React from "react";
import { useImage } from "react-image";
import PropTypes from "prop-types";

import ErrorBoundary from "@ErrorBoundary";

const ImageContainer = (props) => {
  const ImageWrapper = (props) => {
    const {
      props: { url, cn },
    } = props;
    const { src } = useImage({ srcList: url });

    return (
      <>
        <img loading="lazy" src={src} alt={cn} className={cn} />
      </>
    );
  };

  return (
    <ErrorBoundary>
      <ImageWrapper props={props} />
    </ErrorBoundary>
  );
};

export default ImageContainer;

ImageContainer.propTypes = {
  url: PropTypes.string,
  cn: PropTypes.string,
};
