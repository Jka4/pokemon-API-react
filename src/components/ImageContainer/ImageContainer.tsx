import React, { Suspense } from "react";
import { useImage } from "react-image";

import ErrorBoundary from "utils/ErrorBoundary";

interface ImageContainerProps {
  url: string;
  cn?: string;
  fallback?: any
}

const ImageContainer = ({ url, cn, fallback }: ImageContainerProps) => {

  const ImageWrapper = () => {
    const { src } = useImage({ srcList: url });

    return (
      <>
        <img src={src} alt={cn} className={cn} />
      </>
    );
  };

  return (
    <>
      <ErrorBoundary>
        <Suspense fallback={fallback}>
          <ImageWrapper />
        </Suspense>
      </ErrorBoundary>
    </>
  );
};

export default ImageContainer;
