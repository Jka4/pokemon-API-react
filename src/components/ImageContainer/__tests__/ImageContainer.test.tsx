import * as React from 'react';

import ImageContainer, { IMG, FallbackStyled } from '../ImageContainer';

import { mount } from 'enzyme';
import { render } from '@testing-library/react';

describe('<HeaderLine/> ', () => {
  it('renders correctly', () => {
    const container = mount(
      <ImageContainer
        url={'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png'}
        cn="testClass"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('unmount component', () => {
    const { unmount } = render(
      <ImageContainer
        url={'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png'}
        cn="testClass"
      />,
    );

    unmount();
  });

  it('empty url', () => {
    const { unmount } = render(<ImageContainer cn="testClass" />);

    unmount();
  });

  it('image loaded TRUE', () => {
    const imageWrapper = mount(<IMG ready={true} />);
    const fallbackWrapper = mount(<FallbackStyled ready={true} />);

    expect(imageWrapper).toMatchSnapshot();
    expect(fallbackWrapper).toMatchSnapshot();
  });

  it('image loaded FALSE', () => {
    const imageWrapper = mount(<IMG ready={false} />);
    const fallbackWrapper = mount(<FallbackStyled ready={false} />);

    expect(imageWrapper).toMatchSnapshot();
    expect(fallbackWrapper).toMatchSnapshot();
  });
});
