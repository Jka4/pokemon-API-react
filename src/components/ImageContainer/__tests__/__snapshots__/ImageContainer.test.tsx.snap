// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLine/>  image loaded FALSE 1`] = `
<styled.img
  ready={false}
>
  <img
    className="sc-dlfnbm iTRrSd"
  />
</styled.img>
`;

exports[`<HeaderLine/>  image loaded FALSE 2`] = `
<styled.div
  ready={false}
>
  <div
    className="sc-hKgILt kKOCJE"
  />
</styled.div>
`;

exports[`<HeaderLine/>  image loaded TRUE 1`] = `
<styled.img
  ready={true}
>
  <img
    className="sc-dlfnbm gIdFjC"
  />
</styled.img>
`;

exports[`<HeaderLine/>  image loaded TRUE 2`] = `
<styled.div
  ready={true}
>
  <div
    className="sc-hKgILt cubUVT"
  />
</styled.div>
`;

exports[`<HeaderLine/>  renders correctly 1`] = `
<ImageContainer
  cn="testClass"
  url="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
>
  <ErrorBoundary>
    <styled.div
      ready={false}
    >
      <div
        className="sc-hKgILt kKOCJE"
      />
    </styled.div>
    <styled.img
      alt="testClass"
      className="testClass"
      loading="lazy"
      ready={false}
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
    >
      <img
        alt="testClass"
        className="sc-dlfnbm iTRrSd testClass"
        loading="lazy"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png"
      />
    </styled.img>
  </ErrorBoundary>
</ImageContainer>
`;
